torch/csrc/jit/runtime/register_prim_ops_fulljit.cpp

```diff
@@ -127,6 +128,60 @@ RegisterOperators reg(
            };
          },
          aliasAnalysisSpecialCase()),
+     Operator(
+         "prim::AllocateSlab() -> Storage",
+         [](const Node* node) -> Operation {
+           int64_t total_size = node->i(attr::total_size);
+           auto device_type =
+               static_cast<DeviceType>(node->i(attr::device_type));
+           return [total_size, device_type](Stack* stack) {
+             auto allocator = GetAllocator(device_type);
+             auto slab = c10::Storage(
+                 c10::Storage::use_byte_size_t(),
+                 total_size,
+                 allocator,
+                 /*resizable=*/false);
+             push(stack, std::move(slab));
+           };
+         },
+         aliasAnalysisSpecialCase()),
+     Operator(
+         prim::AllocateTensor,
+         [](const Node* node) -> Operation {
+           int64_t size = node->i(attr::size);
+           int64_t offset = node->i(attr::offset);
+           auto type = node->ty(attr::profiled_type)->expect<TensorType>();
+
+           return [offset, size, type](Stack* stack) {
+             c10::Storage slab;
+             pop(stack, slab);
+             uint8_t* start = static_cast<uint8_t*>(slab.data());
+             void* src = static_cast<void*>(start + offset);
+             at::Tensor temp_tensor = at::from_blob(
+                 src,
+                 *type->sizes().concrete_sizes(),
+                 *type->strides().concrete_sizes(),
+                 at::TensorOptions(*type->device()).dtype(*type->scalarType()));
+             temp_tensor.storage().set_nbytes(size);
+             push(stack, std::move(temp_tensor));
+           };
+         },
+         aliasAnalysisSpecialCase()),
+     Operator(
+         "prim::ReleaseSlab(Storage slab, ...) -> ()",
```

Comment:
Hmm, `slab` here could relate to any number of Tensors, and it would be kind of a pain to track all of the dependencies there. I think it probably makes sense just to mark `prim::ReleaseSlab` as an op with side effects for now, which means we can't move any node around it and it won't get DCE'd. 

I'm not sure what a good alternative is. We don't want to set the `slab` as containing the tensors which point to it because that would pessimistically extend their lifetimes. We could maybe set slab to directly alias with these tensors, but it's might be a little weird to have Values of different types alias each other, and will also run into the same extending lifetime issue.

Reply:
I'm not sure what the constraints here are but just fyi I don't know the value (or the sensibility) of trying to do alias analysis on the slices of `slab` that get handed out.

Reply:
Yea, I think you're right. We might want to just make it an invariant that memory planning is the last pass we run, and throw if we see any of these nodes in alias analysis. What do people think cc @Krovatkin @desertfire ?  

What I was trying to avoid is a future pass moving nodes around have dependencies that are not reflected topologically. 
E.g., you can't move prim::ReleaseSlab around another node.

Reply:
More I think about it, all of the logic here is dependent on topology being frozen. we should probably just throw in alias analysis if we see these nodes and enforce invariant this is the final pass

Reply:
@eellison 

>reflected topologically

aren't `uses` relationships exactly topological relationships? all of `AllocateTensor` nodes have a source -> node -> sink relationship with `AllocateSlab` and whichever op consumes the output of `AllocateTensor`?

Reply:
say you have:

```
graph(a, b, c, d):
   y = a + b
   z = c + d
   y1 = y + z   

   y0 = a + c 
   z0 = b + d
   z1 = y0 + z0
    
   return z1 + y1
```


there is no topological constraint about whether to compute `y/z` or `y0/z0` first, however their lifetimes will be implicitly baked into the memory planning scheme. let's say that `y/z` are at offset 0, and 256, and then those same offset/sizes get reused for y0/z0, if you were to move `y0 = a + c` above `z = c + d`, that would overwrite the value of `y`, however, there is nothing reflected topologically that says this is an invalid move

Reply:
@eellison this is a good (very good) observation but if you move `y0 = a + c` above `z = c + d` then you would also be required to move its `AllocateTensor` with it as well right? in which case you could not mistakenly overwrite (unless *I'm* mistaken)